OutOfSchool/OutOfSchool.WebApi.Tests/Validators/CustomAgeAttributeTests.cs

```diff
@@ -0,0 +1,201 @@
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenMinAgeIsNotPositive_ShouldThrowInvalidOperationException()
+    {
+        // Arrange
+        var dateOfBirth = faker.Date.Between(DateTime.MinValue, DateTime.MaxValue);
+
+        // Assert
+        Assert.Throws<InvalidOperationException>(() => new CustomAgeAttribute() { MinAge = -1 }.IsValid(dateOfBirth));
+    }
+
+    [Test]
+    public void IsValid_WhenMaxAgeIsNotPositive_ShouldThrowInvalidOperationException()
+    {
+        // Arrange
+        var dateOfBirth = faker.Date.Between(DateTime.MinValue, DateTime.MaxValue);
+
+        // Assert
+        Assert.Throws<InvalidOperationException>(() => new CustomAgeAttribute() { MaxAge = -1 }.IsValid(dateOfBirth));
+    }
+
+    [Test]
+    public void IsValid_WhenMaxAgeIsLessThanMinAge_ShouldThrowInvalidOperationException()
+    {
+        // Arrange
+        var dateOfBirth = faker.Date.Between(DateTime.MinValue, DateTime.MaxValue);
+
+        // Assert
+        Assert.Throws<InvalidOperationException>(() => new CustomAgeAttribute() { MaxAge = 1, MinAge = 2 }.IsValid(dateOfBirth));
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsLessThanMinAge_ShouldReturnFalse()
+    {
+        // Arrange
+        var dateOfBirth = faker.Date.Past(2, DateTime.UtcNow);
+
+        // Act
+        var isValid = new CustomAgeAttribute() { MinAge = 2 }.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsFalse(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsGreaterThanMaxAge_ShouldReturnFalse()
+    {
+        // Arrange
+        var dateOfBirth = faker.Date.Past(100, DateTime.UtcNow.AddYears(-2));
+
+        // Act
+        var isValid = new CustomAgeAttribute() { MaxAge = 2 }.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsFalse(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsExactlyMinAge_ShouldReturnTrue()
+    {
+        // Arrange
+        var dateOfBirth = DateTime.UtcNow.AddYears(-3);
+
+        // Act
+        var isValid = new CustomAgeAttribute() { MinAge = 3 }.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsExactlyMaxAge_ShouldReturnTrue()
+    {
+        // Arrange
+        var dateOfBirth = DateTime.UtcNow.AddYears(-3);
+
+        // Act
+        var isValid = new CustomAgeAttribute() { MaxAge = 3 }.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsOneDayAfterMinAge_ShouldReturnFalse()
+    {
+        // Arrange
+        var dateOfBirth = DateTime.UtcNow.AddYears(-3).AddDays(1);
+
+        // Act
+        var isValid = new CustomAgeAttribute() { MinAge = 3 }.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsFalse(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsOneDayBeforeMaxAge_ShouldReturnFalse()
```

Comment:
(Test logic) I believe you are misinterpreting age calculation. Please explain why this test says that 3yrs and 1 day old person shouldn't fit `MaxAge=3` condition. 

---

Here is my line of thoughts:

The `CustomAgeAttribute.MaxAge` description says
```
    /// Gets the maximum age of the persom in years.
    /// </summary>
    public int MaxAge { get; init; } = int.MaxValue; 
```

So, `MaxAge` is integer and shows maximum allowed age in years. For me it means, that `MaxAge=3` condition disallows any 4 years old person and older. But every person below 4 years - i.e. 3 yrs and 1 day, as in this test - is allowed.